### PR TITLE
[PUI] Add quick login via url

### PR DESCRIPTION
**Pull request by matmair**
This PR:
- Adds the url based login method (user + pwd) that we use for the demo to PUI
- Switches most tests (all that do not explicitly test user login UI) to use this method as it is easier to compute

### CODE REVIEW

**Comment by matmair** [comment]
ready for review

File: src/frontend/src/pages/Auth/Login.tsx [hunk 1]
```diff
@@ -48,6 +49,17 @@ export default function Login() {
     }
 
     checkLoginState(navigate, location?.state?.redirectFrom, true);
+
+    // check if we got login params (login and password)
+    if (searchParams.has('login') && searchParams.has('password')) {
+      console.log('got login params');
```
Comment (SchrodingersGat) [inline src/frontend/src/pages/Auth/Login.tsx:55]: Please remove this debug msg :)

Comment (matmair) [reply src/frontend/src/pages/Auth/Login.tsx:55]: I thought this might be helpful when debugging the demo instance but I have removed it now

**Comment by matmair** [comment]
@SchrodingersGat this would be ready for review and handy as it cuts down on test run time

**Pull request merged**